libs/blocks/global-footer/global-footer.js

```diff
@@ -28,9 +31,18 @@ const CONFIG = {
 };
 
 class Footer {
-  constructor(footerEl, contentUrl) {
-    this.footerEl = footerEl;
-    this.contentUrl = contentUrl;
+  /**
+   * Footer constructor
+   * @param {Object} config Configuration object for Footer class
+   * @param {Element} config.contentUrl The raw content to decorate to populate the Footer
+   * @param {Element} config.footerEl The placeholder element where the Footer should be rendered
+   * @param {Boolean} [config.useFederatedContent] Whether the Footer loads from a central location
+   */
+  constructor(config) {
+    Object.keys(config).forEach((key) => {
+      this[key] = config[key];
+    });
```

Comment:
I'll revert to using the `this.prop = config.prop` approach, since I'd like for the properties to be visible in the constructor, JSDoc will be removed.